src/components/KYCWall/BaseKYCWall.js

```diff
@@ -89,13 +81,19 @@ class KYCWall extends React.Component {
         // Check to see if user has a valid payment method on file and display the add payment popover if they don't
         if (!PaymentUtils.hasExpensifyPaymentMethod(this.props.cardList, this.props.bankAccountList)) {
             Log.info('[KYC Wallet] User does not have valid payment method');
-            const clickedElementLocation = getClickedElementLocation(event.nativeEvent);
-            const {anchorPositionTop, anchorPositionLeft} = this.getAnchorPosition(clickedElementLocation);
+            let clickedElementLocation = getClickedElementLocation(event.nativeEvent);
+            let position = this.getAnchorPosition(clickedElementLocation);
+            if (this.props.shouldListenForResize) {
+                window.addEventListener('resize', () => {
```

Comment:
Oh, hmm, this is gonna bind a new event listener each time. That can have unintended consequences. The best practice would be to set an event listener once on component init and then clean it up on unmount... 

Here's an example of what that looks like using a class component

https://reactjs.org/docs/hooks-effect.html#example-using-classes-1

cc @rushatgabhane @deetergp

Reply:
hmm, we already added a remove listener when the modal close.

Reply:
Please fix this. Here are the instructions...

1. add event listeners in `componentDidMount()`
2. remove event listeners in `componentWillUnmount()`
3. bind the callbacks they use

Generally speaking even if we are focusing on the web side for consistency we should use the `Dimensions` utility. React native web is able to use this.

https://reactnative.dev/docs/0.66/dimensions#addeventlistener

Reply:
@phivh right now, we're adding a event listener each time the window is resized.
So yeah, let's create a follow-up PR to fix this. What do you think?

Reply:
ok, let me take into this one

Reply:
@marcaaron @deetergp @rushatgabhane created follow-up PR for this one. Please take a look when you guys have a chance.
https://github.com/Expensify/App/pull/7847